pkg/config/config_template.yaml

```
@@ -413,6 +413,130 @@ api_key:
 #
 # secret_backend_timeout: 5
 
+## @param snmp_listener - custom object - optional
+## Creates and schedules an snmp listener to automatically discover your SNMP devices.
+## Discovered devices can then be monitored with the snmp integration by using
+## the auto_conf.yaml file provided by default.
+#
+# snmp_listener:
+
+  ## @param workers - integer - optional - default: 2
+  ## Number of concurrent tasks used to discover SNMP devices. Increasing this value
+  ## will help to discover devices faster but at the cost of increased resource consumption.
+  #
+  # workers: 2
+
+  ## @param discovery_interval - integer - optional - default: 3600
+  ## How often to discover new SNMP devices, in seconds. Decreasing this value
+  ## will help to discover devices faster but at the cost of increased resource consumption.
+  #
+  # discovery_interval: 3600
+
+  ## @param allowed_failures - integer - optional - default: 3
+  ## Number of failed requests to a given SNMP device before removing it from the list of monitored
+  ## devices.
+  ## If a device shuts down, the Agent will stop monitoring it after `discovery_interval * allowed_failures` seconds.
+  #
+  # allowed_failures: 3
+
+  ## @param configs - list - required
+  ## The actual configurations used to discover SNMP devices in various subnets.
+  #
+  # configs: 
+    ## @param network - string - required
+    ## The subnet in CIDR format to scan for SNMP devices.
+    ## All IP addresses in the CIDR range that are not ignored will be scanned.
+    ## For optimal discovery time, be sure to use the smallest network mask
+    ## possible as is appropriate for your network topology.
+    ## Ex: 10.0.0.0/24
+    #
+    # - network: <NETWORK>
+
+    ## @param ignored_ip_addresses - list of strings - optional
+    ## A list of IP addresses to ignore when scanning the network.
+    #
+    # ignored_ip_addresses:
+    #   - <IP_ADDRESS_1>
+    #   - <IP_ADDRESS_2>
+
+    ## @param port - integer - optional - default: 161
+    ## The UDP port to use when connecting to SNMP devices.
+    #
+    # port: 161
+
+    ## @param version - integer - optional - default: 2
+    ## If you are using SNMP v1 set version to 1.
+    ## If you are using SNMP v3 set version to 3.
+    #
+    # version: 2
```

Seems like I didn't submit my comment on this last time.

FWIW it looks like the implementation doesn't just default to using v2c, but rather makes a smart decision based on the usage of `community` (v2c) / `user` (v3).

https://github.com/DataDog/datadog-agent/blob/1fca2472014f8ee014d6e715944dc4b9539a53b5/pkg/snmp/snmp.go#L134-L143

So should we update this to something like…?

```suggestion
    ## @param version - integer - optional
    ## Defaults to 2 if `community` is set, or 3 if `user` is set.
    ## If you are using SNMP v1 set version to 1.
    #
    # version: <VERSION>
```

---

Yes this is definitely more correct but also is an implementation detail that will sound confusing.
What do you think of something like this:

```
    ## @param version - integer - optional - default: <BEST_GUESS>
    ## Set the version of the SNMP protocol. Value can be `1`, `2` or `3`.
    ## If unset, the integration tries to guess the correct version based on other configuration
    ## parameters. i.e if `user` is set, the integration will use SNMP v3.
    #
    # version: <VERSION>
```

---

I like it :+1: